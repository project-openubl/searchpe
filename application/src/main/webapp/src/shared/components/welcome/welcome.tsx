import React from "react";
import {
  EmptyState,
  Title,
  EmptyStateBody,
  Button,
  EmptyStateVariant,
  EmptyStateSecondaryActions,
  EmptyStateIcon,
} from "@patternfly/react-core";
import { RocketIcon } from "@patternfly/react-icons";

export interface WelcomeProps {
  onPrimaryAction: () => void;
}

export const Welcome: React.FC<WelcomeProps> = ({ onPrimaryAction }) => {
  return (
    <EmptyState variant={EmptyStateVariant.small}>
      <EmptyStateIcon icon={RocketIcon} />
      <Title headingLevel="h4" size="lg">
        Bienvenido a Searchpe
      </Title>
      <EmptyStateBody>
        Searchpe te ayuda a consumir los datos proveidos por la SUNAT a travéz
        del 'padrón reducido'. Antes de empezar a consultar datos, necesitas
        crear una <strong>Versión</strong>.
      </EmptyStateBody>
      <Button variant="primary" onClick={onPrimaryAction}>
        Ir a Versiones
      </Button>
      <EmptyStateSecondaryActions>
        Para aprender más visita la
        <a
          target="_blank"
          href="https://project-openubl.github.io"
          rel="noopener noreferrer"
        >
          documentación.
        </a>
      </EmptyStateSecondaryActions>
    </EmptyState>
  );
};
