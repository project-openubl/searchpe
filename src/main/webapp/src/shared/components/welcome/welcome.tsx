import React from "react";
import {
  EmptyState,
  Title,
  EmptyStateBody,
  Button,
  EmptyStateVariant,
  EmptyStateSecondaryActions,
  EmptyStateIcon,
} from "@patternfly/react-core";
import { RocketIcon } from "@patternfly/react-icons";

export interface WelcomeProps {
  onPrimaryAction: () => void;
}

export const Welcome: React.FC<WelcomeProps> = ({ onPrimaryAction }) => {
  return (
    <EmptyState variant={EmptyStateVariant.small}>
      <EmptyStateIcon icon={RocketIcon} />
      <Title headingLevel="h4" size="lg">
        Welcome to Searchpe
      </Title>
      <EmptyStateBody>
        Searchpe helps you to consume the data exposed by the SUNAT through the
        'padr√≥n reducido'. To start searching data you first need to have at
        least one valid <strong>Version</strong>
      </EmptyStateBody>
      <Button variant="primary" onClick={onPrimaryAction}>
        Go to Versions
      </Button>
      <EmptyStateSecondaryActions>
        To learn more, visit the
        <a
          target="_blank"
          href="https://project-openubl.github.io"
          rel="noopener noreferrer"
        >
          documentation.
        </a>
      </EmptyStateSecondaryActions>
    </EmptyState>
  );
};
