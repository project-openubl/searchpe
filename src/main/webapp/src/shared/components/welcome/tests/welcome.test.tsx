import React from "react";
import { shallow, mount } from "enzyme";
import { Button } from "@patternfly/react-core";
import { Welcome } from "../welcome";

describe("Welcome", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<Welcome onPrimaryAction={jest.fn} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("onPrimaryAction should call callback", () => {
    const mockCallback = jest.fn();

    const wrapper = mount(<Welcome onPrimaryAction={mockCallback} />);
    wrapper.find(Button).simulate("click");
    expect(mockCallback.mock.calls.length).toEqual(1);
  });
});
