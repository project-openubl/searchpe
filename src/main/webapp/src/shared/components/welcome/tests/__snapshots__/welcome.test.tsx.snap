// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Renders without crashing 1`] = `
<EmptyState
  variant="small"
>
  <EmptyStateIcon
    icon={[Function]}
  />
  <Title
    headingLevel="h4"
    size="lg"
  >
    Welcome to Searchpe
  </Title>
  <EmptyStateBody>
    Searchpe helps you to consume the data exposed by the SUNAT through the 'padr√≥n reducido'. To start searching data you first need to have at least one valid 
    <strong>
      Version
    </strong>
  </EmptyStateBody>
  <Button
    onClick={[Function]}
    variant="primary"
  >
    Go to Versions
  </Button>
  <EmptyStateSecondaryActions>
    To learn more, visit the
    <a
      href="https://project-openubl.github.io"
      rel="noopener noreferrer"
      target="_blank"
    >
      documentation.
    </a>
  </EmptyStateSecondaryActions>
</EmptyState>
`;
